docs/src/main/asciidoc/chapters/iterator_design.txt

```diff
@@ -145,8 +146,16 @@ alter the internal state of the Iterator.
 
 These methods simply return the current Key-Value pair for this iterator. If `hasTop` returns true,
 both of these methods should return non-null objects. If `hasTop` returns false, it is undefined
-what these methods should return. Multiple calls to these methods should not alter the state
-of the Iterator like `hasTop`.
+what these methods should return. Like `hasTop`, multiple calls to these methods should not alter 
+the state of the Iterator.
+
+When saving a Key or Value from a source iterator's `getTopKey` or `getTopValue` methods
+for use after calling `next` on the source iterator (e.g., when cacheing keys or values
+from the source iterator), it is important to copy the Key or Value into a new object 
+because the source iterator may reuse the Key or Value objects for performance reasons.
```

Comment:
> It's a wordy statement but we need all the clauses

Nothing is simple :)

>  don't like the verb use, as it's vague. I would prefer specifics - ..."which does not modify or cache the key beyond calling next()"

I like this too. Thanks for chiming in, John.

I think your suggestion, Dylan, with John's clarification would be really good.